### ISSUE TITLE
Webpack 5 output.clean deletes loaded resources on rebuild

### ISSUE THREAD

**Issue by andreivictor** [webpack-contrib/html-loader#498]
### Bug report

Using `output.clean: true` with `webpack dev server`, when the project is rebuilt due to source changes, the resources included in HTML are deleted and the browser returns a 404 error.

### Actual Behavior

`index.html` includes some static resources (images), like `<img src="images/header.jpg" />`.
Config works fine in production build - the resource is written in `dist` folder, and included in `index.html` with the proper hashname. It also works when using dev-server, when building for the first time.
However, when the project is rebuilt due to source changes, the image is deleted and the browser returns a 404 error.

If `output.clean` is set to `{dry: true}`, on rebuilt, it logs the following output:
```
LOG from webpack.CleanPlugin
<i> images/header.png will be removed
```

### Expected Behavior

When the project is rebuilt due to source changes, the image should not be deleted.

### How Do We Reproduce?

1. Clone my repo created for debugging this issue: https://github.com/andreivictor/frontend-webpack-boilerplate
2. Run `npm install`
3. Run `npm run dev`
4. Go to `src/js/demo.js` and make some changes so the webpack rebuilds the project
5. Refresh the page

### Please paste the results of `npx webpack-cli info` here, and mention other relevant information
```
System:
    OS: macOS 13.2.1
    CPU: (12) arm64 Apple M2 Max
    Memory: 3.14 GB / 32.00 GB
  Binaries:
    Node: 18.15.0 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 9.5.0 - /usr/local/bin/npm
  Browsers:
    Chrome: 117.0.5938.149
    Safari: 16.3
  Packages:
    babel-loader: ^9.1.3 => 9.1.3
    copy-webpack-plugin: ^11.0.0 => 11.0.0
    css-loader: ^6.8.1 => 6.8.1
    css-minimizer-webpack-plugin: ^5.0.1 => 5.0.1
    html-loader: ^4.2.0 => 4.2.0
    html-webpack-plugin: ^5.5.3 => 5.5.3
    image-minimizer-webpack-plugin: ^3.8.3 => 3.8.3
    postcss-loader: ^7.3.3 => 7.3.3
    sass-loader: ^13.3.2 => 13.3.2
    terser-webpack-plugin: ^5.3.9 => 5.3.9
    webpack: ^5.88.2 => 5.88.2
    webpack-bundle-analyzer: ^4.9.1 => 4.9.1
    webpack-cli: ^5.1.4 => 5.1.4
    webpack-dev-server: ^4.15.1 => 4.15.1
    webpack-merge: ^5.9.0 => 5.9.0
```

**Comment by alexander-akait** [comment]
It was a bug in `html-webpack-plugin`, but already fixed, just set `"html-webpack-plugin": "^5.6.0",` in your `package.json`, sorry for delay and feel free to feedback

**Comment by andreivictor** [comment]
Updating `html-webpack-plugin` to the latest version solved this issue.

Thank you!
